Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoComponentPropertyOverride.cpp

```diff
@@ -73,62 +74,77 @@ namespace AzToolsFramework
+            // Configure paths from focused and top templates. The top template is the source template of the link.
+            AZ::Dom::Path pathToPropertyFromTop =
+                AZ::Dom::Path(PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances, true)) /
+                relativePathFromOwningPrefab;
+            AZ::Dom::Path pathToPropertyFromFocused(PrefabDomUtils::InstancesName);
+            pathToPropertyFromFocused /= climbUpResult.m_climbedInstances.back()->GetInstanceAlias(); // top instance alias
+            pathToPropertyFromFocused /= pathToPropertyFromTop;
+
+            // Get the current state of property value from focused template.
+            const TemplateId focusedTemplateId = link->get().GetTargetTemplateId();
+            const PrefabDom& focusedTemplateDom = m_prefabSystemComponentInterface->FindTemplateDom(focusedTemplateId);
+            PrefabDomPath domPathToPropertyFromFocused(pathToPropertyFromFocused.ToString().c_str());
+            const PrefabDomValue* currentPropertyDomValue = domPathToPropertyFromFocused.Get(focusedTemplateDom);
+
+            if (currentPropertyDomValue)
+            {
+                PrefabDom changePatches;
+                m_instanceToTemplateInterface->GeneratePatch(changePatches, *currentPropertyDomValue, afterStateOfComponentProperty);
+                m_changed = !changePatches.GetArray().Empty();
+            }
+            else
+            {
+                // If the DOM value from focused template is not present, it means the value is new to the template.
+                m_changed = true;
+            }
 
+            if (m_changed)
             {
-                AZ::Dom::Path pathFromFocusedPrefab(PrefabDomUtils::InstancesName);
-                pathFromFocusedPrefab /= climbUpResult.m_climbedInstances.back()->GetInstanceAlias();
-                pathFromFocusedPrefab /= m_overriddenPropertyPathFromFocusedPrefab;
-
-                // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
-                // DOM value pointers can't be relied upon if the original DOM gets modified after pointer creation.
-                PrefabDomPath overriddenPropertyDomPath(pathFromFocusedPrefab.ToString().c_str());
-                const PrefabDomValue* overriddenPropertyDomInTopTemplate = overriddenPropertyDomPath.Get(topTemplateDom);
-
-                
-                PrefabDom overridePatches;
-                if (overriddenPropertyDomInTopTemplate)
+                // Get the default state of property value from top template (direct child prefab of focused template).
+                // Note: We can't get the default state from the owning template because top template might have stored overrides
+                // on the owning instance. Those overrides should be treated as default/base values from the focused template.
```

Comment:
Do you mean to say , 'we can't get the default state from the "focused template"' ? The comment is not clear imo and can be removed altogether as I don't see it clarifying things.

Reply:
I think I can update it. I want to put a note here to address reader's question like "why can't we just use the source template from owning instance as the default state?". The reason is there could be the case that `topInstance != owningInstance` and topInstance has an override edit to change the default state from source.